src/modules/wlr/workspace_manager.cpp

```diff
@@ -210,22 +210,34 @@ WorkspaceGroup::WorkspaceGroup(const Bar &bar, Gtk::Box &box, const Json::Value
 
 auto WorkspaceGroup::fill_persistent_workspaces() -> void {
   if (config_["persistent_workspaces"].isObject() && !workspace_manager_.all_outputs()) {
-    const Json::Value &p_workspaces = config_["persistent_workspaces"];
-    const std::vector<std::string> p_workspaces_names = p_workspaces.getMemberNames();
-
-    for (const std::string &p_w_name : p_workspaces_names) {
-      const Json::Value &p_w = p_workspaces[p_w_name];
-      if (p_w.isArray() && !p_w.empty()) {
-        // Adding to target outputs
-        for (const Json::Value &output : p_w) {
+    const Json::Value &pWorkspaces = config_["persistent_workspaces"];
+    const std::vector<std::string> keys = pWorkspaces.getMemberNames();
+
+    for (const std::string &key : keys) {
+      const Json::Value &value = pWorkspaces[key];
+
+      if (value.isNumeric() && value.asInt() > 0) {
+        // value == amount of workspaces this workspace should have
+        uint32_t monitorId = 0;  // TEMP: how to get monitor ID from output name, that matches Hyprland's ID?
```

Comment:
I'd like some help with this: Hyprland has monitor IDs I can get using `hyprctl`, but I don't know how to do it with native wlroots commands. Does this even make sense for non-Hyprland compositors?

Reply:
For now, I implemented it with hyprctl, as an example of how I can get the IDs. I'd love to hear how I can make it work for other wlroots compositors.